docs/source/contribution.rst

```diff
@@ -261,6 +261,15 @@ Please note the followings when writing the document.
   original one, users should explicitly describe only what is implemented in
   the document.
 
+For changes that modify or add new Cython files, please make sure the pointer types follow these guidelines (`GH-1913 <https://github.com/cupy/cupy/issues/1913>`_).
+
+* Pointers should be ``intptr_t``.
```

Comment:
I think we could mention `void*` as well.
How about something like:
"Pointers should be `void*` if it's only used from within Cython, or `intptr_t` if it's exposed to Python code.

(Please feel free to rephrase)

Reply:
@niboshi I am not sure about the usage of `void*`. It seems in CuPy's Cython codes they're all `ctypedef`'d? Anyway, how about this?
```suggestion
* Pointers should be ``void*`` if only used within Cython, or ``intptr_t`` if exposed to the Python space.
```

Reply:
Applied the above patch, please take a look. Thanks.

Reply:
Thanks, sounds better!
Not all pointers are `ctypedef`s, including raw data pointers.